### PR TITLE
feat: add dropBufferSupport option to improve the performance

### PR DESCRIPTION
**Pull request by luin**
https://github.com/luin/ioredis/wiki/Improve-Performance

### CODE REVIEW

File: test/functional/drop_buffer_support.js [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    });
+    redis.geteval(function (err, res) {
+      expect(err).to.eql(null);
+      expect(res).to.eql('string');
+      redis.disconnect();
+      done();
+    });
+  });
+
+  it('should return strings correctly', function (done) {
```
Comment (shaharmor) [inline test/functional/drop_buffer_support.js]: duplicate from two tests above, did you mean to use `dropBufferSupport: false`

Comment (luin) [inline test/functional/drop_buffer_support.js]: This test makes sure non-binary methods (`get`, `set`) works when `dropBufferSupport` is enabled.

Comment (shaharmor) [inline test/functional/drop_buffer_support.js]: Yeah but there is another test that is exactly the same down below (Look at the last test)

Comment (luin) [inline test/functional/drop_buffer_support.js]: Aha...you're right.
shame-LfFC2erYa5qN2

**Pull request merged**